### PR TITLE
Added Ship Repair Module

### PR DESCRIPTION
**Pull request by 100575155**
Added Ship Repair Module

## Summary
As requested in issue #47, added a new module that will repair/regen the shields of the ship.
Currently uses placeholder icon and prefab.

## Linked Issues
Closes [#47](https://github.com/Liam-Harrison/Celeritas/issues/47)

### CODE REVIEW

File: Assets/Scripts/Scriptables/Systems/ShipShieldRepairSystem.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+using Celeritas.Game.Entities;
+using Celeritas.Scriptables.Interfaces;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace Celeritas.Scriptables.Systems
+{
+
+	[CreateAssetMenu(fileName = "New Ship Shield Repair Modifier", menuName = "Celeritas/Modifiers/Ship Shield Repair")]
+
```
Comment (MistaMishMash) [inline Assets/Scripts/Scriptables/Systems/ShipShieldRepairSystem.cs:11]: Would be nice to have a summary comment of the entire class here, describing generally what it does. (a bit obvious, I know, but nice to have c: )

**Review by MistaMishMash — CHANGES_REQUESTED** [review]
There are some conflicts in this branch, so you will need to rebase it before you can finish your merge. In github desktop from memory, it is branch > rebase current branch. Good idea to double check that everything is still working after the rebase, then you should be good to go c:

I've playtested it, and it looks good! You should add in the icon that Aiden made for you to use for this module though. 
If you need any help putting the module in the ship's inventory to playtest it yourself, please feel free to ask me (Sarah). 
Also, just double checking that it's meant to be a 1x1 module; if that's the right shape, all good there.

File: Assets/Scripts/Scriptables/Systems/ShipShieldRepairSystem.cs (hunk 1 above)
Comment (100575155) [reply Assets/Scripts/Scriptables/Systems/ShipShieldRepairSystem.cs:15]: I've committed the changes and added a summary for the class.
Not sure on what the shape needs to be but we can change it later if required.

**Review by MistaMishMash — APPROVED** [review]
Thanks for updating that icon, and putting in the comment! Sorry about the Github shenanigans, but it all looks great now! C:

**Pull request merged**